fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/metadata/JoinedRecordType.java

```diff
@@ -136,17 +139,37 @@ public CompletableFuture<FDBSyntheticRecord> loadByPrimaryKeyAsync(FDBRecordStor
             } else {
                 futures[i] = store.loadRecordAsync(constituentKey).thenApply(rec -> {
                     if (rec == null) {
```

Comment:
Outside the scope of this PR, `JoinedRecordType` supports outer joins, in theory, but I don't think this code has any handling for the possibility that it is an outer join.

Reply:
I am not sure what kind of support we would need. Can you elaborate (maybe offline) and I'll get a ticket created?

Reply:
Done (opened ticket)